src/publish/mergePublishBranch.ts

```diff
@@ -21,3 +30,18 @@ export function mergePublishBranch(publishBranch: string, branch: string, messag
   }
   return result!;
 }
+
+/**
+ * Calls a specified hook for each package being bumped
+ */
+async function precommitHook(options: BeachballOptions) {
+  const hook = options.hooks?.precommit;
+  if (!hook) {
+    return;
+  }
+
+  const hookRet = hook(options.path);
+  if (hookRet instanceof Promise) {
+    await hookRet;
+  }
```

Comment:
You can await this even if it doesn't return a promise.
```suggestion
  await hook(options.path);
```